tests/publish_public_data_json/test_publish_public_data_json.py

```diff
@@ -0,0 +1,168 @@
+import json
+import pytest
+import subprocess
+
+from google.cloud import bigquery
+from google.cloud import storage
+from google.api_core.exceptions import NotFound
+
+
+class TestPublishJson(object):
+    test_bucket = "moz-fx-data-stage-bigquery-etl"
+    project_id = "moz-fx-data-shar-nonprod-efed"
```

Comment:
Do we have a project for running tests?

Reply:
We do not. It usually makes sense to mock the API and check that files are written correctly using [patch](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch).

I do like the idea of running tests in a test project. It should be acceptable from a privacy perspective if everything we write there comes from our tests.

Reply:
Mocking seems to work well when testing separate methods. I had some troubles testing the individual methods in the script since the script itself cannot be imported as module.

Reply:
Hmm, another reason to refactor some of this. It could be worth it to approach this script as an entry into the `bigquery_etl` package, and the tests run pieces there.